src/feature/hs/hs_service.c

```diff
@@ -4123,9 +4123,13 @@ hs_service_dump_stats(int severity)
         const char *nickname;
 
         intro_node = get_node_from_intro_point(ip);
-        tor_assert(intro_node);
+        if (!intro_node) {
```

Comment:
This first `intro_node` check is actually kinda plausible to happen so perhaps we can add a log here, like we did in the !circ case? We've seen it out in the wild.

I think the nickname check is much less likely to happen so I don't care about a log there, but perhaps we can add one anyway.